### PR TITLE
ES6 lib/add-event-listener

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- es6 `addEventListener`
- re-creates the modules list, I think it had got confused

## What is the value of this and can you measure success?

666

### CODE REVIEW

File: static/src/javascripts/lib/add-event-listener.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    );
+    window.addEventListener('test', null, opts);
+} catch (e) {
+    /* noop */
+}
+
+const addEventListener = (
+    node: Node,
+    name: string,
+    handler: Function,
```
Comment (gustavpursche) [inline static/src/javascripts/lib/add-event-listener.js]: It's kind of obvious, but what do you think about: `Function<Event>`?

File: static/src/javascripts/lib/add-event-listener.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    options: Object = {}
+): void | false => {
```
Comment (gustavpursche) [inline static/src/javascripts/lib/add-event-listener.js]: Just curious: How could this code ever return `false`?

File: static/src/javascripts/lib/add-event-listener.js (hunk 1 above)
Comment (SiAdcock) [inline static/src/javascripts/lib/add-event-listener.js]:
```
(e: Event) => void
```

File: tools/es5to6.json [hunk 3]
```diff
@@ -105,7 +105,7 @@
     "projects/common/modules/commercial/adblock-messages.js"
   ],
   "sndrs": [
-    "lib/add-event-listener.js",
+    "lib/ajax.js",
```
Comment (SiAdcock) [inline tools/es5to6.json]: Is this supposed to rebalance itself every time you complete a conversion?

File: static/src/javascripts/lib/add-event-listener.js (hunk 2 above)
Comment (sndrs) [inline static/src/javascripts/lib/add-event-listener.js]: yeah good catch, i had it returning `false` if node was undefined before, left it accidentally

**Review by SiAdcock — APPROVED** [review]
👍

**Comment by regiskuckaertz** [comment]
What say you rename it `events`, so you can later add other utilities like `delegate`?

File: static/src/javascripts/lib/add-event-listener.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            },
+        }: Object) // https://github.com/facebook/flow/issues/285
+    );
+    window.addEventListener('test', null, opts);
+} catch (e) {
+    /* noop */
+}
+
+const addEventListener = (
+    node: Node,
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/add-event-listener.js]: I think a more accurate type would be `EventTarget` here.

File: static/src/javascripts/lib/add-event-listener.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    name: string,
+    handler: (e: Event) => void,
+    options: Object = {}
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/add-event-listener.js]: Not possible to have defaults for the three options: `passive`, `capture` and `once`?

File: static/src/javascripts/lib/add-event-listener.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+): void => {
+    if (node) {
```
Comment (regiskuckaertz) [inline static/src/javascripts/lib/add-event-listener.js]: I thought that was the whole purpose of Flow, that signatures allow it to run a summative flow analysis and enforce constraints such as mandatory arguments.

Comment (sndrs) [inline static/src/javascripts/lib/add-event-listener.js]: very good point

**Comment by regiskuckaertz** [comment]
✨

**Comment by sndrs** [comment]
all good points @regiskuckaertz – have updated accordingly, look ok?

**Comment by sndrs** [comment]
_wrong button_ 🛑

**Comment by regiskuckaertz** [comment]
@sndrs yes, awesome! 👍

**Comment by gustavpursche** [comment]
❤️ Looks great now!

**Pull request merged**